gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -425,16 +446,18 @@ public void onError(Throwable t) {
         Status status = Status.fromThrowable(t);
         Status.Code statusCode = status.getCode();
         if (TRANSIENT_ERRORS.contains(statusCode)) {
+          logger.atWarning().log(
+              "Caught transient exception for '%s', while uploading to uploadId %s at writeOffset %d."
+                  + " Status: %s",
+              resourceId, uploadId, writeOffset, status.getDescription());
           transientError = t;
         }
         if (transientError == null) {
-          nonTransientError =
-              new IOException(
-                  String.format(
-                      "Caught exception for '%s', while uploading to uploadId %s at writeOffset %d."
-                          + " Status: %s",
-                      resourceId, uploadId, writeOffset, status.getDescription()),
-                  t);
+          nonTransientError = t;
+          logger.atSevere().log(
```

Comment:
Logging is a way of handling the exception, we don't want to log and throw the exception (we are implicitly throwing as the observer awaits & checks the exception)

We should handle one exception only at one layer

Reply:
`getResponseOrThrow` wraps the nonTransientError into IOException. So in `onError` if we do as previous implementation, then we end up wrapping the nonTransientError into IOException twice. Hence removing the wrapping.

Re: logging, we are not throwing any exception in this method, we are merely storing them. Hence added a log to indicate that an exception from the readResponse has occurred at this time.